diff -crBN chat-head/tcl/chat-callback-procs.tcl chat/tcl/chat-callback-procs.tcl
*** chat-head/tcl/chat-callback-procs.tcl	1969-12-31 18:00:00.000000000 -0600
--- chat/tcl/chat-callback-procs.tcl	2009-01-08 12:20:32.000000000 -0600
***************
*** 0 ****
--- 1,34 ----
+ ad_library {
+ 
+     Callback procs for chat
+ 
+     @author Alvaro Rodriguez (alvaro@viaro.net)
+     @creation-date 2008-08-13
+ }
+ 
+ ad_proc -public -callback search::url -impl chat_room {} {
+         returns a url for a chat room to the search package
+ } {
+     set package_id [db_string get_package_id {} -default 0]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}room-enter?room_id=${object_id}&client=ajax"
+ }
+ 
+ ad_proc -public -callback dotlrn::blocks::edit_url -impl chat_room {} {
+         returns a url for the edit action of a chat room to the dotlrn blocks
+ } {
+     set package_id [db_string get_package_id {} -default 0]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}room-edit?room_id=$object_id"
+ }
+ 
+ ad_proc -public -callback dotlrn::blocks::delete_url -impl chat_room {} {
+         returns a url for the delete action of a chat room to the dotlrn blocks
+ } {
+     set package_id [db_string get_package_id {} -default 0]
+     set package_url [apm_package_url_from_id $package_id]
+ 
+     return "${package_url}room-delete?room_id=${object_id}"
+ }
diff -crBN chat-head/tcl/chat-callback-procs.xql chat/tcl/chat-callback-procs.xql
*** chat-head/tcl/chat-callback-procs.xql	1969-12-31 18:00:00.000000000 -0600
--- chat/tcl/chat-callback-procs.xql	2009-01-08 12:21:00.000000000 -0600
***************
*** 0 ****
--- 1,29 ----
+ <?xml version="1.0"?>
+ 
+ <queryset>
+ 
+   <fullquery name="callback::search::url::impl::chat_room.get_package_id">
+     <querytext>
+         select context_id
+         from acs_objects
+           where object_id = :object_id
+     </querytext>
+   </fullquery>
+ 
+   <fullquery name="callback::dotlrn::blocks::edit_url::impl::chat_room.get_package_id">
+     <querytext>
+         select context_id
+         from acs_objects
+           where object_id = :object_id
+     </querytext>
+   </fullquery>
+ 
+   <fullquery name="callback::dotlrn::blocks::delete_url::impl::chat_room.get_package_id">
+     <querytext>
+         select context_id
+         from acs_objects
+           where object_id = :object_id
+     </querytext>
+   </fullquery>
+ 
+ </queryset>
diff -crBN chat-head/www/room-edit.tcl chat/www/room-edit.tcl
*** chat-head/www/room-edit.tcl	2009-01-08 12:18:59.000000000 -0600
--- chat/www/room-edit.tcl	2009-01-08 12:19:05.000000000 -0600
***************
*** 6,11 ****
--- 6,12 ----
      @creation-date March 26, 2006
  } {
      room_id:integer,optional
+     {page_instance_id 0}
  } 
  
  permission::require_permission -object_id [ad_conn package_id] -privilege chat_room_edit
***************
*** 79,84 ****
--- 80,89 ----
        #-1 The Public
        chat_user_grant $room_id -2
      }
+ ## Learning Content: Insert the activity_id when the message is created
+         callback -catch -impl content learning_content::insert_object -name $pretty_name -item_id $page_instance_id -activity_id $room_id
+ ###
+ 
  } -edit_request {
      if {[catch {db_1row room_info {
          select pretty_name, description, moderated_p, archive_p, active_p, auto_flush_p, auto_transcript_p
***************
*** 98,100 ****
--- 103,108 ----
      ad_script_abort    
  }
  
+ ## Learning Content: Extend the form to send the page_instance_id for content activities
+     callback -catch -impl content learning_content::extend_form -name "edit-room" -page_instance_id $page_instance_id
+ ###
